scintillometry/pulse_folding.py

```diff
@@ -0,0 +1,107 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import operator
+import warnings
+
+import numpy as np
+import astropy.units as u
+from astropy.utils import lazyproperty
+
+from .base import TaskBase
+
+
+__all__ = ['Fold']
+
+
+class Fold(TaskBase):
+    """ Fold pulses.
+
+    Parameters
+    ----------
+    ih : task or `baseband` stream reader
+        Input data stream, with time as the first axis.
+
+    fold_periods : int
```

Comment:
I think the sample_rate of the result should be u.cycle as well. However in the TaskBase line:
```            
 sample_rate_ratio = (ih.sample_rate / sample_rate).to(1).value
```
will complain about the unit. Since the previous sample rate is in the unit of time.

Reply:
Ah, now I understand the problem. I fear I may have tried to do too much in `TaskBase.__init__` - it doesn't seem to scale well. It may be better to move that calculation back down to `Channelize` (or make a separate class that doesn't presume so much).